File: Data Structures & Algorithms/top-k-elements-in-list/submission-0.py
from typing import List

class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        freq = {}

        for num in nums:
            freq[num] = freq.get(num, 0) + 1

        sorted_nums = sorted(freq.items(), key=lambda x: x[1], reverse=True)

        res = []

        for i in range(k):
            res.append(sorted_nums[i][0])

        return res

File: Data Structures & Algorithms/house-robber/submission-0.cpp
class Solution {
public:
    int rob(vector<int>& nums) {
        int n=(int)nums.size();
        vector<int>dp(n+1);
        dp[0]=0;
        dp[1]=nums[0];
        for(int i=2;i<=n;i++){
            dp[i]=max((dp[i-1]),(dp[i-2]+nums[i-1]));
            cout<<i<<" "<<dp[i-1]<<" "<<dp[i-2]<<" "<<nums[i-1]<<endl;
        }
        return dp[n];
    }
};
